execution_workflows/Aptardi/main.nf

```diff
@@ -0,0 +1,138 @@
+#!/usr/bin/env nextflow
+
+// write help message here
+if (params.help) {
+	
+	    log.info"""
+	    ==============================================
+	    APAeval Aptardi execution workflow
+	    ==============================================
+	    Usage:
+	    Run the pipeline with default parameters:
+	    nextflow run main.nf -profile docker --input samplesheet.csv
+	    Mandatory arguments:
+                --input		samplesheet
+	    Flags:
+                --help			Display this message
+	    """.stripIndent()
+
+	exit 1
+}
+
+// define parameters (add more parameters if necessary)
+if (params.input) { ch_input = file(params.input, checkIfExists: true) } else { exit 1, "Samplesheet file not specified!" }
+if (params.fasta) { ch_fasta = file(params.fasta, checkIfExists: true) } else { exit 1, "FASTA file not specified!" }
+if (params.gtf)   { ch_gtf   = file(params.gtf, checkIfExists: true)   } else { exit 1, "GTF file not specified!" }
+ch_output_bed = params.output_bed
+if (params.aptardi_model) { ch_aptardi_model = file(params.aptardi_model, checkIfExists: true) } // if not specify, the default one will be used
+if (params.aptardi_scale) { ch_aptardi_scale = file(params.aptardi_scale, checkIfExists: true) } // if not specify, the default one will be used
+
+/*
+ * Check input files indicated on the samplesheet to make sure that the input files are valid input for the downstream processes
+ */
+process CHECK_SAMPLESHEET {
+    tag "$samplesheet"
+    publishDir "${params.outdir}/pipeline_info", mode: params.publish_dir_mode
+
+    input:
+    path samplesheet from ch_input
+    
+    output:
+    path "*reformat.csv" into ch_samplesheet_reformat
+    
+    script:
+    """
+    check_samplesheet.py $samplesheet samplesheet_reformat.csv
+    """
+}
+
+def get_sample_info(LinkedHashMap sample) {
+    return [ sample.sample, sample.bam ]
+}
+
+// create input channel for aptardi
+ch_samplesheet_reformat
+    .splitCsv(header:true, sep:',')
+    .map { get_sample_info(it) }
+    .into { ch_input
+            ch_stringtie_input }
+
+if (params.use_stringtie2_gtf){
+    /*
+     * Generate GTF file from running STRINGTIE2
+     */
+    process STRINGTIE2 {
+        tag "$sample"
+        label 'process_medium'
+        publishDir "${params.outdir}/aptardi", mode: params.publish_dir_mode
+
+        input:
+        tuple val(sample), path(bam) from ch_stringtie_input
+        path gtf from ch_gtf
+        path fasta from ch_fasta
+
+        output:
+        tuple val(sample), path("stringtie.merged.gtf"), path(fasta) into ch_stringtie_gtf
+
+        script:
+        """
+        stringtie -G $gtf -o ${sample}.stringtie.gtf $bam
+        stringtie --merge ${sample}.stringtie.gtf -G $gtf -o stringtie.merged.gtf
```

Comment:
I changed this according to your comment, and there are no Aptardi sites found in the Stringtie only GTF:
![Screenshot from 2022-01-06 08-34-27](https://user-images.githubusercontent.com/41866052/148309715-4df46476-ccc6-4751-9cc3-74688195643c.png)

Reply:
I was able to get a successful test run for the `use_stringtie2_ gtf` with a subsampled BAM file.

![aptardi-stringtie2-gtf-success](https://user-images.githubusercontent.com/49978382/150505347-5d287b09-0c24-4dca-b476-1e4685f784af.png)


The identification BED looks good too (though there are duplicate positions, will address this in a separate comment).
![aptardi-stringtie2-id-bed](https://user-images.githubusercontent.com/49978382/150506943-878b0d4d-fa96-4484-bce9-f604abb97e01.png)


Happy to share the test data I used (the subsampled BAM is 46MB, if you add the subsetted GTF & FASTA file it's ~250MB) maybe on a Google Drive if you want to test locally yourselves @faricazjj @yuukiiwa ? The BAM is from publicly available data so no worries on that front.

Reply:
I've come barging in on this PR with a commit of my own, apologies for stepping on your toes @yuukiiwa! I tested the following out locally and had no errors so thought I'd save you the time.

Commit: d43c93a

Successful run with my modifications (StringTie2 pipeline):

![aptardi-id-bed-my-changes](https://user-images.githubusercontent.com/49978382/150514478-2abc7781-cfd9-4bbd-b337-fd8306490cc6.png)


[Per review comment here](https://github.com/iRNA-COSI/APAeval/pull/174#discussion_r789516498), I see duplicate coordinates reported in the BED file with the StringTie2 pipeline. I suspect this could also happen with other runs, so I think we should add a quick step to remove duplicates. The new commit generates a BED file without duplicates for the StringTie run (top = BED with previous script, bottom = new script):



![aptardi dups in bed 2022-01-21 10-45-23](https://user-images.githubusercontent.com/49978382/150514030-c2c0ffd7-2204-4406-a775-f291d953aceb.png)

I also think the error message that happens if you have no aptardi transcripts ([my comment](https://github.com/iRNA-COSI/APAeval/pull/174#pullrequestreview-834333143), @yuukiiwa 's [comment](https://github.com/iRNA-COSI/APAeval/pull/174#discussion_r779227309)) is quite cryptic. I added a check to see if there are no Aptardi transcripts in the input GTF file. If so, a clearer error message is raised:

![aptardi my id-bed nicer error message 2022-01-21 11-09-11](https://user-images.githubusercontent.com/49978382/150517248-d9102e5b-0df5-4946-b867-d28837c79589.png)



I tried out a separate option where the message is printed to stderr & an empty output file is generated (so workflow completes successfully). I opted not to include this in the commit for now as I couldn't seem to find the stderr log message (probably my nextflow inexperience), so an empty output file without this clarification could be confusing. Happy to change that though if either of you (@yuukiiwa @faricazjj) deem it appropriate.
 
![aptardi my id bed changes output empty](https://user-images.githubusercontent.com/49978382/150517448-87648606-cddb-4e9c-ae40-526639de047f.png)

Reply:
Thank you so much!!